lib/idx/authenticate.ts

```diff
@@ -0,0 +1,65 @@
+/* eslint-disable complexity */
+import { AuthTransaction } from '../tx';
+import { OktaAuth, AuthorizeOptions, IdxTransactionMeta, RemediationValues } from '../types';
+import { interact } from './interact';
+import { introspect } from './introspect';
+import { remediate } from './remediate';
+
+export async function authenticate(authClient: OktaAuth, options: AuthorizeOptions): Promise<AuthTransaction> {
+  let { state, scopes, interactionHandle } = options;
+
+  if (interactionHandle) {
+    // resume existing transaction, not supported yet
+    throw new Error('Unexpectedly found a saved interaction handle. Check storageManager config');
+  }
+
+  const interactResponse = await interact(authClient, { state, scopes, interactionHandle });
+  const { stateHandle } = interactResponse;
+  if (!interactionHandle) {
+    interactionHandle = interactResponse.interactionHandle;
+  }
+
+  let idxResponse = await introspect(authClient, { stateHandle });
```

Comment:
can we just use the `interactResponse` from the last step as the `idxResponse`?

Reply:
Hmm. maybe so. Looks like `idx-js` automatically does the `introspect` call. good catch!